File: models/project.go
package models

import (
	"time"
)

type Project struct {
	Id          int
	Name        string
	Description string `sql:"type:text"`
	CreatorId   int    `gorm:"foreignKey"`
	CreatedAt   time.Time
	Users       []User `gorm:"many2many:user_projects;"`
	Creator     User
}

File: models/project.js
var mongoose = require('mongoose');
var bluebird = require('bluebird');

var projectSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  description: {
    type: String,
    required: true
  },
  path: {
    type: String,
    required: true
  },
  creator: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  github: {
    type: String
  },
  members: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
  }]
});

var project = mongoose.model('Project', projectSchema);

bluebird.promisifyAll(project);
bluebird.promisifyAll(project.prototype);

module.exports = project;